nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -221,4 +244,17 @@ protected WriteConcern getWriteConcern(final ProcessContext context) {
         }
         return writeConcern;
     }
+
+    protected void writeBatch(String payload, ProcessContext context, ProcessSession session) {
+        FlowFile flowFile = session.create();
+        flowFile = session.write(flowFile, new OutputStreamCallback() {
+            @Override
+            public void process(OutputStream out) throws IOException {
+                out.write(payload.getBytes("UTF-8"));
```

Comment:
session.importFrom method can save you from callback. String payload can be converted to byte Input Stream and provided to importFrom method.

Reply:
That's a good catch. I made that change.